mythril/laser/ethereum/natives.py

```diff
@@ -53,30 +54,29 @@ def ecrecover(data):
     return o
 
 
-def sha256(data):
+def sha256(data: Union[bytes, str]) -> bytes:
     try:
         data = bytes(data)
     except TypeError:
         raise NativeContractException
     return hashlib.sha256(data).digest()
 
 
-def ripemd160(data):
+def ripemd160(data: Union[bytes, str]) -> List:
```

Comment:
for data variable for the functions namely, this one, sha256, ecrecover should have the same type right? As all are being called from the same function.

Reply:
Interesting finding. Regarding the function signatures, `ripemd160` returns a zero-padded list, while `sha256` returns the raw bytestring digest. You can also check that out in ipython or similar:
```
In [11]: ripemd160("foo".encode())[:16]
Out[11]: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 66, 207, 162, 17]

In [12]: sha256("foo".encode())
Out[12]: b',&\xb4kh\xff\xc6\x8f\xf9\x9bE<\x1d0A4\x13B-pd\x83\xbf\xa0\xf9\x8a^\x88bf\xe7\xae'
```
The output of both could be unified as `typing.Iterable`, but I tried to stay as close to the implementation as possible when coming up with the hints.

Reply:
It might be good to make them return something same as either of the 4 functions are called each time, let's add that as a Todo in that case.

And also the data arg for identity should also have the same type as that of the other 3 functions mentioned above.

Reply:
I have added the missing hint for the identity function. It also now throws a `NativeContractException` if the input cannot be converted to bytes - just like the other two functions. I also made sure `ripemd160` also always returns `bytes` after the padding has been added. So now all functions exhibit the same behaviour, take the same inputs and yield the same outputs. :+1: